### PR TITLE
[Freshdesk] Improvements/Refactoring

### PR DESCRIPTION
**Pull request by tianjing-li**
### What's being changed:

- Refactors client code into `client.py` and FreshdeskClient class
- Improves query building logic
- Clarifies limitations in README

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):
- Tested locally

### CODE REVIEW

File: freshdesk/.env-template [hunk 1]
```diff
@@ -1,4 +1,3 @@
 FRESHDESK_API_KEY=
 FRESHDESK_DOMAIN_NAME=
-FRESHDESK_TICKET_PARAMETER=
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply freshdesk/.env-template:3]: Did not mean to remove, thx for catching

File: freshdesk/provider/client.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        response = requests.get(search_url, auth=self.basic_auth, params=params)
+
+        if response.status_code != 200:
+            message = response.text or f"Error: HTTP {response.status_code}"
+            raise UpstreamProviderError(message)
+
+        return response.json()["results"]
+
+
+def get_client():
```
Comment (EugeneLightsOn) [inline freshdesk/provider/client.py:43]: Is it possible to implement the `global client` pattern in this connector? Freshdesk API doesn't support OAuth.

**Review by EugeneLightsOn — APPROVED** [review]
Looks good to me. I only have one small comment, and please also review Walter's comment.

File: freshdesk/provider/client.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+from . import UpstreamProviderError
+
+
+class FreshdeskClient:
+    DEFAULT_TICKET_PARAMETERS = ["tag"]
+
+    def __init__(self, api_key, domain, parameters=None):
+        self.base_url = f"https://{domain}/api/v2"
+        # Freshdesk uses Basic Auth with this specific format, using the API key
+        self.basic_auth = (api_key, "X")
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply freshdesk/provider/client.py:13]: Basic auth you send a (user, password) pair, I find it strange myself but for Freshdesk the (user, password) they require is the (api_key, "X") instead. Will add a doc link

[the remark this answers is not in the document]
Comment (tianjing-li) [reply freshdesk/provider/client.py]: Done!

File: freshdesk/provider/client.py (hunk 2 above)
Comment (tianjing-li) [reply freshdesk/provider/client.py:46]: Done

**Pull request merged**